packages/flutter/lib/src/cupertino/radio.dart

```diff
@@ -0,0 +1,312 @@
+const double _kCupertinoFocusColorOpacity = 0.80;
+const double _kCupertinoFocusColorBrightness = 0.69;
+const double _kCupertinoFocusColorSaturation = 0.835;
+
+/// A macOS style radio button.
+///
+/// Used to select between a number of mutually exclusive values. When one radio
+/// button in a group is selected, the other radio buttons in the group cease to
+/// be selected. The values are of type `T`, the type parameter of the
+/// [CupertinoRadio] class. Enums are commonly used for this purpose.
+///
+/// The radio button itself does not maintain any state. Instead, selecting the
+/// radio invokes the [onChanged] callback, passing [value] as a parameter. If
+/// [groupValue] and [value] match, this radio will be selected. Most widgets
+/// will respond to [onChanged] by calling [State.setState] to update the
+/// radio button's [groupValue].
+///
+/// {@tool dartpad}
+/// Here is an example of CupertinoRadio widgets wrapped in CupertinoListTiles.
+///
+/// The currently selected character is passed into `groupValue`, which is
+/// maintained by the example's `State`. In this case, the first [CupertinoRadio]
+/// will start off selected because `_character` is initialized to
+/// `SingingCharacter.lafayette`.
+///
+/// If the second radio button is pressed, the example's state is updated
+/// with `setState`, updating `_character` to `SingingCharacter.jefferson`.
+/// This causes the buttons to rebuild with the updated `groupValue`, and
+/// therefore the selection of the second button.
+///
+/// ** See code in examples/api/lib/cupertino/radio/radio.0.dart **
+/// {@end-tool}
+///
+/// See also:
+///
+///  * [CupertinoSlider], for selecting a value in a range.
+///  * [CupertinoCheckbox] and [CupertinoSwitch], for toggling a particular value on or off.
+///  * [Radio], the Material Design equivalent.
+///  * <https://developer.apple.com/design/human-interface-guidelines/components/selection-and-input/toggles/>
+class CupertinoRadio<T> extends StatefulWidget {
+  /// Creates a macOS-styled radio button.
+  ///
+  /// The radio button itself does not maintain any state. Instead, when the
+  /// radio button is selected, the widget calls the [onChanged] callback. Most
+  /// widgets that use a radio button will listen for the [onChanged] callback
+  /// and rebuild the radio button with a new [groupValue] to update the visual
+  /// appearance of the radio button.
+  ///
+  /// The following arguments are required:
+  ///
+  /// * [value] and [groupValue] together determine whether the radio button is
+  ///   selected.
+  /// * [onChanged] is called when the user selects this radio button.
+  const CupertinoRadio({
+    super.key,
+    required this.value,
+    required this.groupValue,
+    required this.onChanged,
+    this.toggleable = false,
+    this.activeColor,
+    this.inactiveColor,
+    this.fillColor,
+    this.focusColor,
+    this.focusNode,
+    this.autofocus = false,
+  });
+
+  /// The value represented by this radio button.
```

Comment:
Add something about "if this equals the [groupValue], then this radio button will appear selected."